students/alex_skrn/lesson06/mailroom.py

```diff
@@ -0,0 +1,273 @@
+    """Return a float -- the last donation of the given donor."""
+    return donors[name][-1]
+
+
+def get_donations(name):
+    """Return a list of the specified donor's donations."""
+    return donors[name]
+
+
+def get_total_given(name):
+    """Return total amount of donations for the given donor."""
+    return sum(get_donations(name))
+
+
+def sort_donors_by_total():
+    """Return a list of donor names sorted by total donations, max to min."""
+    donors_L = list(donors.items())
+    donors_sorted = sorted(donors_L, key=lambda x: sum(x[1]), reverse=True)
+    sorted_donor_names = [item[0] for item in donors_sorted]
+
+    return sorted_donor_names
+
+
+def print_donor_names():
+    """Print existing donor names on screen in alphabetical order."""
+    donors_L = sorted(donors.keys())
+    num = len(donors_L)
+    donors_S = (("\n" + ", ".join(["{}"] * num)).format(*donors_L))
+    print(donors_S)
+
+
+def get_email(name, amount):
+    """Return a str containing a thank-you email."""
+    email_text = ("""\nDear {},\n
+                  \nI would like to thank you for your donation of ${}.\n
+                  \nWe appreciate your support.\n
+                  \nSincerely,\n
+                  \nThe Organization\n
+                  """)
+    return email_text.format(name, amount)
+
+
+def create_report():
+    """Create and print a report."""
+    report = ""
+    title_line_form = "\n{:<26}{:^3}{:>13}{:^3}{:>13}{:^3}{:>13}\n"
+    title_line_text = ('Donor Name', '|', 'Total Given', '|',
+                       'Num Gifts', '|', 'Average Gift'
+                       )
+    report += title_line_form.format(*title_line_text)
+    report += str('- ' * 38)
+    form_line = "\n{:<26}{:>3}{:>13}{:>3}{:>13}{:>3}{:>13}"
+    for name in sort_donors_by_total():
+        total = get_total_given(name)
+        num_gifts = len(get_donations(name))
+        mean = round((total / num_gifts), 2)
+        report += (form_line.format(str(name), '$', str(total), ' ',
+                                    str(num_gifts), '$', str(mean)
+                                    )
+                   )
+    report += "\n"
+    print(report)
+
+
+# PRINT ON SREEN A THANK YOU LETTER TO SOMEONE WHO JUST MADE A DONATION
+def input_donation(name):
+    """Obtain the donation amount from the user."""
+    prompt_amount = "Enter the donation amount or 0 to abort > "
+    while True:
+        try:
+            donation_amount = float(input(prompt_amount))
+        except ValueError:
+            print("Input must be a number")
+            continue
+        else:
+            if donation_amount == 0.0:
+                return False
+            elif donation_amount < 0:
+                print("Input must not be negative")
+                continue
+            else:
+                add_donation(name, donation_amount)
+                return True
+
+
+def old_donor_interaction():
+    """Call any_donor() for the old donor functionality."""
+    any_donor()
```

Comment:
No real need to create this function since i just calls another function and there is no logic in it.